bridge-deploy/bridge_deploy/cli.py

```diff
@@ -0,0 +1,276 @@
+import click
+
+# from web3 import Web3, EthereumTesterProvider
+
+from deploy_tools.cli import (
+    jsonrpc_option,
+    keystore_option,
+    gas_option,
+    gas_price_option,
+    nonce_option,
+    auto_nonce_option,
+    connect_to_json_rpc,
+    retrieve_private_key,
+    get_nonce,
+)
+from deploy_tools.deploy import build_transaction_options
+from deploy_tools.files import validate_and_format_address, InvalidAddressException
+
+from bridge_deploy.home import (
+    deploy_home_bridge_contract,
+    initialize_home_bridge_contract,
+)
+
+
+def validate_address(ctx, param, value):
+    # TODO:
+    # Potentially reformat this to deploy-tools? (Copypasta from validator-set-deploy)
+    # Refactoring might be a thing.
+    """This function must be at the top of click commands using it"""
+    try:
+        return validate_and_format_address(value)
+    except InvalidAddressException as e:
+        raise click.BadParameter(
+            f"The address parameter is not recognized to be an address: {value}"
+        ) from e
+
+
+validator_set_address_option = click.option(
+    "--validator-set-address",
+    help='The address of the validator set contract, "0x" prefixed string',
+    type=str,
+    required=True,
+    callback=validate_address,
+    metavar="VALIDATOR_SET_ADDRESS",
+    envvar="VALIDATOR_SET_ADDRESS",
+)
+
+block_reward_address_option = click.option(
+    "--block-reward-address",
+    help='The address of the block reward contract, "0x" prefixed string',
+    type=str,
+    required=True,
+    callback=validate_address,
+    metavar="BLOCK_REWARD_ADDRESS",
+    envvar="BLOCK_REWARD_ADDRESS",
+)
+
+owner_address_option = click.option(
+    "--owner-address",
+    help='The address of the contract owner, "0x" prefixed string',
+    type=str,
+    required=True,
+    callback=validate_address,
+    metavar="OWNER_ADDRESS",
+    envvar="OWNER_ADDRESS",
+)
+
+home_daily_limit_option = click.option(
+    "--home-daily-limit",
+    help="The daily transfer limit for the home bridge in WEI",
+    type=int,
+    default=30_000_000_000_000_000_000_000_000,
+)
+
+home_max_per_tx_option = click.option(
+    "--home-max-per-tx",
+    help="The maximum transfer limit for one transaction in WEI",
+    type=int,
+    default=1_500_000_000_000_000_000_000_000,
+)
+
+home_min_per_tx_option = click.option(
+    "--home-min-per-tx",
+    help="The minimum transfer limit for one transaction in WEI",
+    type=int,
+    default=500_000_000_000_000_000,
+)
+
+home_gas_price_option = click.option(
+    "--home-gas_price",
+    help="The initial gas price on the home network in WEI",
+    type=int,
+    default=1_000_000_000,
+)
+
+required_block_confirmations_option = click.option(
+    "--required-block-confirmations",
+    help="The number of blocks to wait for finality",
+    type=int,
+    default=4,
+)
+
+foreign_daily_limit_option = click.option(
```

Comment:
I don't think we need this option as the limit should either not exist (if we write our own burning bridge contract) or be set to zero.

Reply:
0 is not allowed by the contract, but we can set it to 1 WEI.
I wanted to make it configurable, but we can also hard-code the foreign options.

Reply:
Changed.

Reply:
It's only restricted in `initialize`, after that the owner can set it to 0.

Reply:
Set the tx limit to 0 and the daily limit to 1 for now. Will update with a separate tx.

Reply:
Done.